bubblesort.rb

```diff
@@ -0,0 +1,55 @@
+def bubble_sort(x)
+    x.each do |item|
+    
+        index = 0
+    while index < x.size - 1
+    
+        if x[index] > x[index + 1]
+     
+            x[index], x[index + 1] = x[index + 1], x[index]
+    
+    
+        end 
+    
+        index+=1
+    
+    end
+    
+end
+
+    
+x
+
+end
+
+puts bubble_sort([4,3,78,2,0,2])
+
+
+
+
+def bubble_sort_by(arr)
+    arr.each do |_item|
+        index = 0
+        while index < arr.size - 1
+            if yield(arr[index], arr[index+1])
+                arr[index], arr[index+1] = arr[index+1], arr[index]
+            end
+            index+=1
+        end 
+    end
+
+    arr
+end
```

Comment:
Let's make `bubble_sort_by` work as shown on the [Odin assignment page](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-1).

```ruby
 bubble_sort_by(["hi","hello","hey"]) do |left,right|
    left.length - right.length
 end

# Result is currently ["hi", "hello", "hey"] instead of  ["hi", "hey", "hello"]
```

Instead of using `each` I would use a `while` loop to sort the array until all elements are sorted.